New package: ComputingRegions v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: ComputingRegions
- Repository: https://github.com/JuliaAlgebra/ComputingRegions.jl
- Created by: @PBrdng
- Version: v0.1.0
- Commit: b939a914a9469cec02892c2e593c91b5aeba04ec
- Reviewed by: @PBrdng
- Reference: https://github.com/JuliaAlgebra/ComputingRegions.jl/commit/b939a914a9469cec02892c2e593c91b5aeba04ec#commitcomment-146430741
<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd6247a0ef5af7b67099bfbc68e55a35e681c4b9b463dc6e3f0a48d258d4ac346ff667e2fac9ebbce27c3fbefda18f7c1c37a713d7f3a3ada479b7883f4f7d3f0802ce8e3e3821501abd766af7f899a35817d9716f866b66738daf5d1493fc2296b9717e13e1a9e331ea41219f080f0081fd0a3427b6388324ef21cfa20097854a0c8fe5a4846d6e7c42c632257a2ad3845334f2ea2eaae0e3651fa58e2d07d3347d391ebfe662415912a136569acc035fc5a8bd7111d583264455cef87f8a3778d -->

goerz commented:
This seems like a bit of a weird name. I'd read it as "Regions of Computing", which I don't think is what you're intending. A present-participle verb as part of a package name would be unusual: I think package names should work grammatically with "using <name>".

JuliaRegistrator committed:
New package: ComputingRegions v0.1.0

UUID: 6d1d4b64-24a0-4125-b8e7-5a4e0a5017ad
Repo: https://github.com/JuliaAlgebra/ComputingRegions.jl.git
Tree: 2f64acf0c00576933186a321b6430192d7c188d6

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a

PBrdng commented:
Would "RegionsComputing" be more suitable?

goerz commented:
My recommendation would be something more specific, like `HyperplaneRegions`. I'm not in on the math, so that's probably not an optimal name, but you get the picture: describe what kind of region the package deals with, if that's possible with one or two words. The more mathematically specialized it is, the more specific the package name should be. We very much encourage long package names (three or even four words), if necessary. As the [guidelines](https://pkgdocs.julialang.org/v1/creating-packages/#Package-naming-guidelines) say, "Err on the side of clarity, even if clarity seems long-winded to you."

goerz commented:
> In our case, "regions" is the complement of a hypersurface. There is not really a mathematical term for this, unfortunetaly.

`HyperComplementRegions`? But again, you're probably in a much better position than me to come up with a suitable name.

PBrdng commented:
> My recommendation would be something more specific, like `HyperplaneRegions`. I'm not in on the math, so that's probably not an optimal name, but you get the picture: describe what kind of region the package deals with, if that's possible with one or two words. The more mathematically specialized it is, the more specific the package name should be. We very much encourage long package names (three or even four words), if necessary. As the [guidelines](https://pkgdocs.julialang.org/v1/creating-packages/#Package-naming-guidelines) say, "Err on the side of clarity, even if clarity seems long-winded to you."



> > In our case, "regions" is the complement of a hypersurface. There is not really a mathematical term for this, unfortunetaly.
> 
> `HyperComplementRegions`? But again, you're probably in a much better position than me to come up with a suitable name.

What about `HyperplaneComplementRegions`? Would that work?

goerz commented:
Yes, sorry, I didn't see your edit

> Would "HypersurfaceComplement" or "ComplementOfHypersurface" be more suitable?

`HypersurfaceComplement` and `HyperplaneComplementRegions` or any variation like that would be totally fine

PBrdng commented:
> Yes, sorry, I didn't see your edit
> 
> > Would "HypersurfaceComplement" or "ComplementOfHypersurface" be more suitable?
> 
> `HypersurfaceComplement` and `HyperplaneComplementRegions` or any variation like that would be totally fine

Okay, I will discuss with my coauthors what name to choose.

PBrdng commented:
You can close this. See #114840
